sparse_ho/criterion.py

```diff
@@ -306,3 +346,80 @@ def get_val_grad(
             grad = None
 
         return val, grad
+
+
+class CrossVal():
+    def __init__(self, X, y, Model, cv=5, test_size=0.33, max_iter=1000):
+        """
+        Parameters
+        ----------
+        X : {ndarray, sparse matrix} of (n_samples, n_features)
+            Data
+        y : {ndarray, sparse matrix} of (n_samples)
+            Target
+        Model: class from Model (e.g. Lasso or Sparse logistic regression)
+        cv: can be an integer or predefined folds
+        test_size: float
+        max_iter: int
+            Maximal number of iteration for the soa solver
+        """
+        self.X = X
+        self.y = y
+        self.dict_crits = {}
+        self.val_test = None
+        self.rmse = None
+
+        # init dict of models
+        if isinstance(cv, int):
```

Comment:
cv=int defaults to KFold(int) so this branching is not correct. What do you here is a ShuffleSplit in sklearn

Reply:
you mean that if cv is an int I should do this :
`kf = KFold(n_splits=5, shuffle=True, random_state=42)`